packages/@aws-cdk/pipelines/lib/pipeline.ts

```diff
@@ -314,7 +315,15 @@ export class CdkPipeline extends CoreConstruct {
     return flatMap(this._pipeline.stages, s => s.actions.filter(isDeployAction));
   }
 
-  private* validateDeployOrder(): IterableIterator<string> {
+  private updateArtifactBucketTags(): void {
+    const accounts = this.stackActions.map(
+      (deploymentAction) => Arn.parse(deploymentAction.actionRole.roleArn).account,
+    ).filter(Boolean).sort() as string[];
+
+    Tags.of(this._pipeline.artifactBucket).add('aws-cdk:${Qualifier}:read-from-account', `,${accounts.join(',')},`);
```

Comment:
This `Qualifier` is not going to do anything.

By the way, this is not the only artifact bucket. There may also be replication buckets for cross-region deployments in the CodePipeline.